concent_api/core/tests/test_task_result_upload_finished.py

```diff
@@ -0,0 +1,138 @@
+                subtask_id=self.subtask_id,
+            )
+        )
+
+    def test_that_result_upload_finished_should_change_result_upload_finished_field_on_subtask(self):
+        assert self.subtask.result_upload_finished is False
+
+        result_upload_finished(self.subtask.subtask_id)  # pylint: disable=no-value-for-parameter
+
+        self.subtask.refresh_from_db()
+        self.assertTrue(self.subtask.result_upload_finished)
+
+    def test_that_result_upload_finished_should_raise_exception_for_non_existing_subtask(self):
+        with self.assertRaises(Subtask.DoesNotExist):
+            result_upload_finished(self._get_uuid('3'))  # pylint: disable=no-value-for-parameter
+
+    def test_that_result_upload_finished_should_raise_exception_for_subtask_with_reported_state(self):
+        self.subtask.state = Subtask.SubtaskState.FORCING_REPORT.name  # pylint: disable=no-member
+        self.subtask.save()
+
+        with self.assertRaises(SubtaskStatusError):
+            result_upload_finished(self.subtask.subtask_id)  # pylint: disable=no-value-for-parameter
+
+    def test_that_result_upload_finished_should_raise_exception_for_subtask_with_failed_state(self):
+        self.task_id = self._get_uuid('2')
+        self.subtask_id = self._get_uuid('2')
+
+        self.subtask = store_subtask(
+            task_id=self.task_id,
+            subtask_id=self.subtask_id,
+            provider_public_key=self.PROVIDER_PUBLIC_KEY,
+            requestor_public_key=self.REQUESTOR_PUBLIC_KEY,
+            state=Subtask.SubtaskState.FAILED,
+            next_deadline=None,
+            task_to_compute=self._get_deserialized_task_to_compute(
+                task_id=self.task_id,
+                subtask_id=self.subtask_id,
+            ),
+            report_computed_task=self._get_deserialized_report_computed_task(
+                subtask_id=self.subtask_id,
+            )
+        )
+
+        with mock.patch('logging.Logger.info') as log_info:
+            result_upload_finished(self.subtask.subtask_id)  # pylint: disable=no-value-for-parameter
+
+        self.assertEqual(log_info.call_count, 2)
+
+    def test_that_result_upload_finished_should_raise_exception_for_subtask_with_other_than_forcing_result_transfer_state(self):
+        self.subtask.state = Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name  # pylint: disable=no-member
+        self.subtask.save()
+
+        with mock.patch('logging.Logger.warning') as log_warning:
+            result_upload_finished(self.subtask.subtask_id)  # pylint: disable=no-value-for-parameter
+
+        log_warning.assert_called()
+
+
+class CoreResultUploadFinishedTaskTransactionTest(TransactionTestCase):
+
+    multi_db = True
+
+    def setUp(self):
+        super().setUp()
+
+        (self.PROVIDER_PRIVATE_KEY, self.PROVIDER_PUBLIC_KEY) = generate_ecc_key_pair()
+        (self.REQUESTOR_PRIVATE_KEY, self.REQUESTOR_PUBLIC_KEY) = generate_ecc_key_pair()
```

Comment:
Please move it `TransactionTestCase`

Reply:
`TransactionTestCase` is taken directly from Django, I can't modify it.

Reply:
O, I didn't know I need to read about it ;)

Reply:
Check the import :)